tt_metal/impl/device/device.hpp

```diff
@@ -257,67 +229,58 @@ class Device {
 
     bool using_slow_dispatch() const;
 
+    // Next cluster of methods could be private but they are actively used by DevicePool to maintain device lifecycle
     // Checks that the given arch is on the given pci_slot and that it's responding
     // Puts device into reset
     bool initialize(const uint8_t num_hw_cqs, size_t l1_small_size, size_t trace_region_size, tt::stl::Span<const std::uint32_t> l1_bank_remap = {}, bool minimal = false);
-    void initialize_cluster();
-    std::unique_ptr<Allocator> initialize_allocator(size_t l1_small_size, size_t trace_region_size, tt::stl::Span<const std::uint32_t> l1_bank_remap = {});
-    void initialize_build();
-    void initialize_device_kernel_defines();
     void build_firmware();
-    void initialize_device_bank_to_noc_tables(const HalProgrammableCoreType &core_type, CoreCoord virtual_core);
-    void initialize_firmware(const HalProgrammableCoreType &core_type, CoreCoord virtual_core, launch_msg_t *launch_msg, go_msg_t* go_msg);
     void reset_cores();
     void initialize_and_launch_firmware();
     void init_command_queue_host();
     void init_command_queue_device();
     void initialize_synchronous_sw_cmd_queue();
+    void update_dispatch_cores_for_multi_cq_eth_dispatch();
+
+    // review: 17 arguments, wtf
+    // only relies on device id, relies more on cluster
```

Comment:
I believe this is already planned to be removed by @tt-dma ?
This is used by the old FD init flow, and David has already merged a new init scheme. This is only still around to make sure there are no issues with the new flow before it gets removed.

Reply:
Do we have an issue # for cleaning up the old path?

Reply:
thats ok, I'll check on this with David later. no critical/urgent